src/catkin_pkg/condition.py

```diff
@@ -57,8 +57,27 @@ def _evaluate(parse_results, context):
         # return literal value
         return parse_results
 
+    assert len(parse_results) >= 3
+
     # recursion
-    assert len(parse_results) == 3
+
+    if (len(parse_results) % 2) != 0:
```

Comment:
What happens in the `else` case? The following code requires the `parse_results` to have three items which might not be the case anymore.

Reply:
https://github.com/nepython/catkin_pkg/blob/033ab12454d88f33f7f5a68cee7e99fa6ac6c3a6/src/catkin_pkg/condition.py#L60-L64
This code requires that there be odd number of arguments in parse result greater than 3.
eg. 3, 5, 7,..

Reply:
These code snippet is supposed to allow only odd number of arguments greater than 3 for parse_results 
eg. 3,5,7,.. as this is expected.

Reply:
I understand what the condition does. But if the length is e.g. 4 it doesn't go into the `if` block. But the code below the if block is expecting exactly 3 and won't work correctly with a length of 4, 6, 8, etc.

Reply:
The parse_results should look like:-
If one `or` / `and` is present `[['foo', '<=', 'bar'], 'or', ['bar', '>=', 'baz']] `args = **3**
If two `or` / `and` are present `[['foo', '<=', 'bar'], 'and', ['bar', '<', 'baz'], 'or', ['foo', '>', 'bar']]` args = **5**
and so on...
So, I don't think 4, 6, 8, etc. are even possible.

Reply:
Then please readd an assertion which ensures that this precondition is met.